unique.php

```diff
@@ -2931,6 +3091,107 @@ function ewww_image_optimizer_install_pngout() {
+ */
+function ewww_image_optimizer_install_svgcleaner() {
+	if ( ! extension_loaded( 'zlib' ) || ! class_exists( 'PharData' ) ) {
+		$download_error = __( 'zlib or phar extension missing from PHP', 'ewww-image-optimizer' );
+	}
+	$os_chmod  = true;
+	$os_binary = 'svgcleaner';
+	if ( PHP_OS === 'Linux' ) {
+		$arch_type = 'x86_64';
+		if ( ewww_image_optimizer_function_exists( 'php_uname' ) ) {
+			$arch_type = php_uname( 'm' );
+		}
+		$os_string = 'linux_' . $arch_type;
+	} elseif ( PHP_OS === 'Darwin' ) {
+		$os_string = 'macos';
+	} elseif ( PHP_OS === 'WINNT' ) {
+		$os_chmod  = false;
+		$os_string = 'win32';
+		$os_binary = 'svgcleaner.exe';
+	}
+	$latest    = '0.9.5';
+	$tool_path = trailingslashit( EWWW_IMAGE_OPTIMIZER_TOOL_PATH );
+	if ( empty( $download_error ) ) {
+		$download_result = download_url( 'https://github.com/RazrFalcon/svgcleaner/releases/download/v' . $latest . '/svgcleaner_' . $os_string . '_' . $latest . '.tar.gz' );
+		if ( is_wp_error( $download_result ) ) {
+			$download_error = $download_result->get_error_message();
+		} else {
+			if ( ! ewwwio_check_memory_available( filesize( $download_result ) + 1000 ) ) {
+				$download_error = __( 'insufficient memory available for installation', 'ewww-image-optimizer' );
+			} else {
+				$tmpname  = current( explode( '.', $download_result ) );
+				$tmpname .= '-' . uniqid() . '.tar.gz';
+				rename( $download_result, $tmpname );
+				$download_result = $tmpname;
+
+				$pkg_gzipped     = new PharData( $download_result );
+				$pkg_tarball     = $pkg_gzipped->decompress();
+				$download_result = $pkg_tarball->getPath();
+				$pkg_tarball->extractTo(
+					EWWW_IMAGE_OPTIMIZER_BINARY_PATH,
+					'svgcleaner',
+					true
+				);
+				if ( ewwwio_is_file( EWWW_IMAGE_OPTIMIZER_BINARY_PATH . $os_binary ) ) {
+					if ( ! rename( EWWW_IMAGE_OPTIMIZER_BINARY_PATH . $os_binary, $tool_path . $os_binary ) ) {
+						if ( empty( $download_error ) ) {
+							$download_error = __( 'could not move svgcleaner', 'ewww-image-optimizer' );
+						}
+					}
+					if ( $os_chmod && ! chmod( $tool_path . $os_binary, 0755 ) ) {
+						if ( empty( $download_error ) ) {
+							$download_error = __( 'could not set permissions', 'ewww-image-optimizer' );
+						}
+					}
+					if ( PHP_OS === 'WINNT' ) {
+						$pkg_version = ewww_image_optimizer_tool_found( '"' . $tool_path . $os_binary . '"', 's' );
+					} else {
+						$pkg_version = ewww_image_optimizer_tool_found( ewww_image_optimizer_escapeshellarg( $tool_path ) . $os_binary, 's' );
+					}
+				} else {
+					$download_error = __( 'extraction of files failed', 'ewww-image-optimizer' );
+				}
+			}
+		}
+	}
+	if ( is_string( $download_result ) && is_writable( $download_result ) ) {
+		unlink( $download_result );
+	}
+	if ( ! empty( $pkg_version ) ) {
+		$sendback = add_query_arg( 'ewww_svgcleaner', 'success', remove_query_arg( array( 'ewww_svgcleaner', 'ewww_error' ), wp_get_referer() ) );
```

Comment:
Now, I could be missing something, but I couldn't find any place where we set the disable_svgcleaner option to false. Nor do we give the user that ability, except via using a constant/override like we have with PNGOUT.
My goal with svgcleaner though, is that it isn't hidden so much as PNGOUT, so I don't think they should have to explicitly set the disable_svgcleaner option.
Instead, I think this is a perfect spot to set disable_svgcleaner to false, since we have successfully installed it and verified that it works.

Reply:
ok, make sense. I thought you want to have it done similarly to pngout.